testing/endtoend/evaluators/beaconapi_evaluators/beacon_api.go

```diff
@@ -307,150 +497,145 @@ func withCompareBeaconAPIs(beaconNodeIdx int) error {
+			return errors.Wrap(err, "failed to unmarshal prysm ssz")
 		}
-		if err := blockP.UnmarshalSSZ(sszrspP); err != nil {
-			return errors.Wrap(err, "prysm ssz error")
-		}
-
-		if len(blockP.Signature) == 0 || len(blockL.Signature) == 0 || hexutil.Encode(blockP.Signature) != hexutil.Encode(blockL.Signature) {
-			return fmt.Errorf("prysm response %v does not match lighthouse response %v",
-				blockP,
-				blockL)
+	} else if finalizedEpoch >= helpers.BellatrixE2EForkEpoch && finalizedEpoch < helpers.CapellaE2EForkEpoch {
+		b := &ethpb.SignedBeaconBlockBellatrix{}
+		if err := b.UnmarshalSSZ(ssz); err != nil {
+			return errors.Wrap(err, "failed to unmarshal prysm ssz")
 		}
 	} else {
-		blockP := &ethpb.SignedBeaconBlockBellatrix{}
-		blockL := &ethpb.SignedBeaconBlockBellatrix{}
-		if err := blockL.UnmarshalSSZ(sszrspL); err != nil {
-			return errors.Wrap(err, "lighthouse ssz error")
-		}
-		if err := blockP.UnmarshalSSZ(sszrspP); err != nil {
-			return errors.Wrap(err, "prysm ssz error")
+		b := &ethpb.SignedBeaconBlockCapella{}
+		if err := b.UnmarshalSSZ(ssz); err != nil {
+			return errors.Wrap(err, "failed to unmarshal prysm ssz")
 		}
+	}
 
-		if len(blockP.Signature) == 0 || len(blockL.Signature) == 0 || hexutil.Encode(blockP.Signature) != hexutil.Encode(blockL.Signature) {
-			return fmt.Errorf("prysm response %v does not match lighthouse response %v",
-				blockP,
-				blockL)
-		}
+	// verify that dependent root of proposer duties matches block header
+	blockHeaderData := requests["/beacon/headers/{param1}"]
+	header, ok := blockHeaderData.prysmResps["json"].(*beacon.GetBlockHeaderResponse)
+	if !ok {
+		return errJsonCast
 	}
-	blockheaderData := beaconPathsAndObjects["/beacon/headers/{param1}"]
-	prysmHeader, ok := blockheaderData.prysmResps["json"].(*beacon.GetBlockHeaderResponse)
+	dutiesData := requests["/validator/duties/proposer/{param1}"]
+	duties, ok := dutiesData.prysmResps["json"].(*validator.GetProposerDutiesResponse)
 	if !ok {
-		return errors.New("failed to cast type")
+		return errJsonCast
+	}
+	if header.Data.Root != duties.DependentRoot {
+		return fmt.Errorf("header root %s does not match duties root %s ", header.Data.Root, duties.DependentRoot)
 	}
-	proposerdutiesData := beaconPathsAndObjects["/validator/duties/proposer/{param1}"]
-	prysmDuties, ok := proposerdutiesData.prysmResps["json"].(*validator.GetProposerDutiesResponse)
+
+	// get first peer returned from /node/peers and use it's ID to test the /node/peers/{peer_id} endpoint
+	peersData := requests["/node/peers"]
+	pPeers, ok := peersData.prysmResps["json"].(*node.GetPeersResponse)
 	if !ok {
-		return errors.New("failed to cast type")
+		return errJsonCast
 	}
-	if prysmHeader.Data.Root != prysmDuties.DependentRoot {
-		genesisTime, err := strconv.ParseUint(genesisData.Data.GenesisTime, 10, 64)
-		if err != nil {
-			return errors.Wrapf(err, "could not parse genesis time")
-		}
-		fmt.Printf("current slot: %v\n", slots.CurrentSlot(genesisTime))
```

Comment:
This code seemed totally irrelevant

Reply:
I wanted to see when things failed so i printed the slot there.